kbase-extension/static/kbase/js/widgets/function_output/kbaseReportView.js

```diff
@@ -442,206 +377,140 @@ define([
-                                                        displayData[k].description +
-                                                        '</td>'
-                                                )
-                                        );
-                                    }
-                                    $tblDiv.append($objTable);
-                                    reportRowEvents();
-                                } else {
-                                    const $tbl = $(
-                                        '<table cellpadding="0" cellspacing="0" border="0" style="width: 100%; margin-left: 0px; margin-right: 0px;">'
-                                    ).addClass('table table-bordered table-striped');
-                                    $tblDiv.append($tbl);
-
-                                    const tblSettings = {
-                                        paginationType: 'full_numbers',
-                                        displayLength: numPerPage,
-                                        dom: 'ft<ip>',
-                                        sorting: [[0, 'asc']],
-                                        columns: [
-                                            {
-                                                title: '<b>Created Object Name</b>',
-                                                data: 'name',
-                                                width: '30%',
-                                            },
-                                            { title: '<b>Type</b>', data: 'type', width: '20%' },
-                                            { title: '<b>Description</b>', data: 'description' },
-                                        ],
-                                        data: [],
-                                        language: {
-                                            search: 'Search: ',
-                                            emptyTable: 'No created objects.',
-                                        },
-                                    };
-                                    const objTable = $tbl.dataTable(tblSettings);
-                                    objTable.fnAddData(displayData);
-                                    reportRowEvents();
-                                    objTable.on('draw.dt', () => {
-                                        reportRowEvents();
-                                    });
-                                }
-                            })
-                            .catch((error) => {
-                                console.error(error);
-                            });
-                    }
-                }
+            if (this.options.showCreatedObjects) {
+                const objectWidget = OutputWidget.make();
+                const objectsNode = document.createElement('div');
+                this.$mainPanel.append(objectsNode);
+                renderPromises.push(objectWidget.start({
```

Comment:
Right now, there's only one Promise added here, but maybe more later if we come back to it, or eventually pop the other panels out into individual widgets. This just lets us capture and properly consume the Promise that the objectWidget creates.